src/ImgixTag.js

```diff
@@ -7,45 +7,55 @@ var ImgixTag = (function() {
     this.settings = opts || {};
 
     if (!this.el) {
-      throw new Error('ImgixTag must be passed a DOM element.');
-    }
-
-    if (this.el.hasAttribute('ix-initialized') && !this.settings.force) {
-      return;
-    }
-
-    this.ixPathVal = el.getAttribute(this.settings.pathInputAttribute);
-    this.ixParamsVal = el.getAttribute(this.settings.paramsInputAttribute);
-    this.ixSrcVal = el.getAttribute(this.settings.srcInputAttribute);
-    this.ixHostVal =
-      el.getAttribute(this.settings.hostInputAttribute) || this.settings.host;
-
-    if (this.ixPathVal && !this.ixHostVal) {
-      throw new Error(
-        'You must set a value for `imgix.config.host` or specify an `ix-host` attribute to use `ix-path` and `ix-params`.'
-      );
-    }
+      console.warn('ImgixTag must be passed a DOM element.');
+    } else {
+      if (this.el.hasAttribute('ix-initialized') && !this.settings.force) {
+        return;
+      }
 
-    this.baseParams = this._extractBaseParams();
-    this.baseUrl = this._buildBaseUrl();
-    this.baseUrlWithoutQuery = this.baseUrl.split('?')[0];
+      this.ixPathVal = el.getAttribute(this.settings.pathInputAttribute);
+      this.ixParamsVal = el.getAttribute(this.settings.paramsInputAttribute);
+      this.ixSrcVal = el.getAttribute(this.settings.srcInputAttribute);
+      this.ixHostVal =
+        el.getAttribute(this.settings.hostInputAttribute) || this.settings.host;
+
+      if (this.ixPathVal && !this.ixHostVal) {
+        console.warn(
+          'You must set a value for `imgix.config.host` or specify an `ix-host` attribute to use `ix-path` and `ix-params`.'
+        );
+      } else if (
+        typeof this.ixPathVal === 'string' &&
+        this.ixPathVal.length == 0
+      ) {
+        console.warn('`ix-path` cannot accept a value of empty string ""');
+      } else if (
+        typeof this.ixSrcVal === 'string' &&
+        this.ixSrcVal.length == 0
+      ) {
+        console.warn('`ix-src` cannot accept a value of empty string ""');
```

Comment:
Same comment for these three validations--I think it reads better if they're each checked in a standalone `if` block, with a warning and an empty `return;` inside.